### PR TITLE
[Finishes #113071279] Add jwt consolidation to third-party auth

### PR DESCRIPTION
**Pull request by bryomckim**
@thegoodson Kindly review and merge

### CODE REVIEW

File: server/services/auth/google-auth.js [hunk 1]
```diff
@@ -22,15 +23,37 @@
...
+                  email: user.email
+                }, secretKey, {
+                  expiresIn: '86400h'
+                });
+
+                user.token = token;
+                user.google_auth_token = null;
+                user.save();
                 // return user
+                req.session.user =  user.token;
```
Comment (thegoodson) [inline server/services/auth/google-auth.js]: We don't use the sessions anymore.
We should remove it.

File: server/services/auth/google-auth.js [hunk 2]
```diff
@@ -22,15 +23,37 @@
...
+                    var token = jwt.sign({
+                      id: user.id,
+                      email: user.email
+                    }, secretKey, {
+                      expiresIn: '86400h'
+                    });
+
+                    user.token = token;
+                    user.save();
+                    user.google_auth_token = null;
```
Comment (thegoodson) [inline server/services/auth/google-auth.js]: Any reason for removing the 3rd party token before saving the user?

File: server/services/auth/google-auth.js (hunk 1 above)
Comment (bryomckim) [inline server/services/auth/google-auth.js]: @thegoodson I used `session.user` to save the token in the request object for use in the success redirect callback.

File: server/services/auth/google-auth.js (hunk 2 above)
Comment (bryomckim) [inline server/services/auth/google-auth.js]: @thegoodson I thought it was a security concern sending the 3rd party token. I may be wrong. What do you think?

File: server/services/auth/google-auth.js (hunk 1 above)
Comment (thegoodson) [inline server/services/auth/google-auth.js]: The token can be passed directly into the `req` object like we do with our `decoded` object from the `Users.authenticate` middleware.

File: server/services/auth/google-auth.js (hunk 2 above)
Comment (thegoodson) [inline server/services/auth/google-auth.js]: It's  a security risk, as well as the whole database of any web application, the token will be required for OAuth1.0 transactions with the user's 3rd party account when the need arises. It's safe to have it stored.

**Pull request merged**